xgi/classes/reportviews.py

```diff
@@ -319,6 +330,32 @@ def __repr__(self):
         """
         return f"{self.__class__.__name__}({dict(self)})"
 
+    def _get_degrees(self):
+        degrees = dict()
+        if self._order is None:
+            if self._weight is None:
+                for id, nbrs in self._ids.items():
+                    degrees[id] = len(nbrs)
+            else:
+                for id, nbrs in self._ids.items():
+                    degrees[id] = sum(
+                        self._id_attrs[dd].get(self._weight, 1) for dd in nbrs
+                    )
+        else:
+            if self._weight is None:
+                for id, nbrs in self._ids.items():
+                    degrees[id] = len(
+                        [i for i in nbrs if len(self._neighbor_ids[i]) == self._order]
```

Comment:
```suggestion
                        i for i in nbrs if len(self._neighbor_ids[i]) == self._order
```

Reply:
I get a "object of type 'generator' has no len()" error

Reply:
oh lol duh!